src/components/NewAnalysisModal.js

```diff
@@ -1,37 +1,106 @@
 import _ from 'lodash/fp'
 import { Fragment, useState } from 'react'
-import { div, h, hr, img } from 'react-hyperscript-helpers'
-import { spinnerOverlay } from 'src/components/common'
+import { div, h, h2, hr, img, span } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, Select, spinnerOverlay, WarningTitle } from 'src/components/common'
+import Dropzone from 'src/components/Dropzone'
+import { icon } from 'src/components/icons'
 import ModalDrawer from 'src/components/ModalDrawer'
 import { NewGalaxyModalBase } from 'src/components/NewGalaxyModal'
 import { NewRuntimeModalBase } from 'src/components/NewRuntimeModal'
-import { tools } from 'src/components/notebook-utils'
+import { analysisNameInput, analysisNameValidator, getDisplayName, getTool, notebookData, tools } from 'src/components/notebook-utils'
 import TitleBar from 'src/components/TitleBar'
 import galaxyLogo from 'src/images/galaxy-logo.png'
 import jupyterLogoLong from 'src/images/jupyter-logo-long.png'
 import rstudioLogo from 'src/images/rstudio-logo.svg'
+import { Ajax } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
-import { withErrorReporting } from 'src/libs/error'
+import { reportError, withErrorReporting } from 'src/libs/error'
+import { FormLabel } from 'src/libs/forms'
+import { getCurrentApp, getCurrentRuntime, isRuntimeDeletable } from 'src/libs/runtime-utils'
+import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
+import validate from 'validate.js'
 
-//TODO: title id in titleBar and aria-labelled-by in withModalDrawer
 
 const titleId = 'new-analysis-modal-title'
+const NEW_ANALYSIS_MODE = 'NEW_ARTIFACT'
+const NEW_ENVIRONMENT_MODE = 'NEW_ENVIRONMENT'
```

Comment:
```suggestion
const newAnalysisMode = Symbol('new-artifact')
const newEnvironmentMode = Symbol('new-environment')
```
We owe you some notes about this convention in our contribution guide. Using `Symbol` isn't that important (but it's a nice technique). We are trying to avoid SCREAMING_SNAKE_CASE.